include/beast/http/write.hpp

```diff
@@ -18,25 +18,59 @@
 namespace beast {
 namespace http {
 
-/** Write a HTTP/1 message to a stream.
+/** Write a HTTP/1 message on a stream.
 
-    @param stream The stream to send the message on.
+    This function is used to write a message to a stream. The call
+    will block until one of the following conditions is true:
 
-    @param msg The message to send.
+    @li The entire message is sent.
+
+    @li An error occurs.
+
+    This operation is implemented in terms of one or more calls
+    to the stream's `write_some` function.
+
+    The implementation will automatically perform chunk encoding if
+    the contents of the message indicate that chunk encoding is required.
+    If the semantics of the message indicate that the connection should
+    be closed after the message is sent, the error returns from this
+    function will be `boost::asio::error::eof`.
+
+    @param stream The stream to which the data is to be written.
+    The type must support the @b `SyncWriteStream` concept.
 
-    @throws boost::system::error code on failure.
+    @param msg The message to write.
+
+    @throws boost::system::error Thrown on failure.
 */
 template<class SyncWriteStream,
     bool isRequest, class Body, class Headers>
 void
 write(SyncWriteStream& stream,
     message_v1<isRequest, Body, Headers> const& msg);
 
-/** Write a HTTP/1 message to a stream.
+/** Write a HTTP/1 message on a stream.
 
-    @param stream The stream to send the message on.
+    This function is used to write a message to a stream. The call
+    will block until one of the following conditions is true:
 
-    @param msg The message to send.
+    @li The entire message is sent.
+
+    @li An error occurs.
+
+    This operation is implemented in terms of one or more calls
+    to the stream's `write_some` function.
+
+    The implementation will automatically perform chunk encoding if
+    the contents of the message indicate that chunk encoding is required.
+    If the semantics of the message indicate that the connection should
+    be closed after the message is sent, the error returns from this
```

Comment:
Moving this comment to the correct line: "the error return*ed* from this function"